addons/purchase/models/account_invoice.py

```diff
@@ -157,10 +157,8 @@ def _anglo_saxon_purchase_move_lines(self, i_line, res):
             # calculate and write down the possible price difference between invoice price and product price
             for line in res:
                 if line.get('invl_id', 0) == i_line.id and reference_account_id == line['account_id']:
-                    valuation_price_unit = i_line.product_id.uom_id._compute_price(i_line.product_id.standard_price, i_line.uom_id)
```

Comment:
I think that line should stay since it won't be covered by line 

https://github.com/odoo/odoo/pull/32180/files#diff-0f1ebb3dd2d9f9251c966ddc2226def0R194

Reply:
I'm fairly certain this line is necessary. This is column 21, and the `valuation_price_unit` set at line 194 is at column 25. Therefore, if you don't verify the condition of line 161, you will end up with an undefined variable at line 209 since it is a condition aligned on column 21.